contracts/0.4.24/nos/NodeOperatorsRegistry.sol

```diff
@@ -699,4 +767,158 @@ contract NodeOperatorsRegistry is INodeOperatorsRegistry, IsContract, AragonApp
         KEYS_OP_INDEX_POSITION.setStorageUint256(keysOpIndex + 1);
         emit KeysOpIndexSet(keysOpIndex + 1);
     }
+
+    function _trimUnusedNodeOperatorKeys(uint256 _nodeOperatorId) internal returns (uint64 trimmedKeys) {
+        uint64 totalSigningKeys = operators[_nodeOperatorId].totalSigningKeys;
+        uint64 usedSigningKeys = operators[_nodeOperatorId].usedSigningKeys;
+
+        // write only if update is needed
+        if (totalSigningKeys != usedSigningKeys) {
+            trimmedKeys = totalSigningKeys - usedSigningKeys;
+            operators[_nodeOperatorId].totalSigningKeys = usedSigningKeys; // discard unused keys
+            emit NodeOperatorTotalKeysTrimmed(_nodeOperatorId, trimmedKeys);
+        }
+    }
+
+    function _setTotalActiveKeys(uint64 _newTotalActiveKeys) internal {
+        keysUsageStats.totalActiveKeys = _newTotalActiveKeys;
+    }
+
+    function _updateTotalAvailableKeysCount() internal {
+        uint256 totalAvailableKeysCount;
+        uint256 activeNodeOperatorsCount = getNodeOperatorsCount();
+        for (uint256 i = 0; i < activeNodeOperatorsCount; ++i) {
+            totalAvailableKeysCount += _getNodeOperatorAvailableKeysCount(i);
+        }
+        keysUsageStats.totalAvailableKeys = to64(totalAvailableKeysCount);
```

Comment:
Shouldn't we have an event to cover every storage modification?
We don't have one for `keysUsageStats` at all as far as I see.

Reply:
fixed